ray-operator/config/samples/ray_v1alpha1_rayservice.yaml

```diff
@@ -17,7 +17,8 @@ spec:
           working_dir: "https://github.com/ray-project/test_dag/archive/78b4a5da38796123d9f9ffff59bab2792a043e95.zip"
         deployments:
           - name: MangoStand
-            num_replicas: 1
```

Comment:
This is used to ensure that both the head and worker have at least one Serve replica each. If a Pod doesn't have a Ray Serve replica, it will not have the Proxy Actor since Ray 2.8.0. Hence, the readiness probe will fail and thus the RayServiceAddCREvent can't converge. See [this line](https://github.com/ray-project/kuberay/blob/024aaefce1540ab810a972150714aa165c9648fb/tests/test_sample_rayservice_yamls.py#L70) for more details.

Reply:
I think we should add this as a comment in the test to prevent it from becoming tribal knowledge (for example, if later some future maintainer changes `num_replicas` to 1 for whatever reason, the test will start flaking and nobody will remember why). Ideally we should just make the test robust to both cases if it's not too hard.

However if this issue with the readiness probe is just temporary, then it's fine.

Reply:
> Ideally, we should just make the test robust to both cases if it's not too hard.

Agree. I will expose more information in the RayService status later so that the RayServiceAddCREvent doesn't need to wait until all Pods are ready.

Reply:
Update https://github.com/ray-project/kuberay/pull/1823/commits/6a2047d4388423edbb82617a81fbecd4bede3ae8